packages/frontity/src/utils/slot-and-fill/slot.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import connect from "@frontity/connect";
+import { Connect, Package } from "../../../types";
+import useFills from "./use-fills";
+
+interface SlotProps {
+  /*
+   * The name of the Slot.
+   * The user of this Slot will have to specify this name in order to insert a Fill component.
+   */
+  name: string;
+
+  /*
+   * Any data that you might want to pass to the Fill.
+   * Normally used for passing route data fetched in the parent component.
+   */
+  data?: any;
+
+  /*
+   * Any other prop.
+   * The theme can specify other props and they will be passed down to the Fill.
+   */
+  [key: string]: any;
+}
+
+type SlotType = React.FC<Connect<Package, SlotProps>>;
+
+/**
+ * A React component that renders a Fill to fulfill a Slot and Fill pattern.
+ */
+const Slot: SlotType = ({ name, state, children, ...slotProps }) => {
```

Comment:
The properties `actions` and `libraries` are being included in `slotProps` and passed to `Fill` as props later.

This is wrong, isn't it? :thinking:

I guess the way of passing `state`, `actions` and `libraries` would be to connect the `Fill` component before exporting it.

Reply:
> I guess the way of passing state, actions and libraries would be to connect the Fill component before exporting it.
yeah, you're right that this is probably not what we want. 

Those `...slotProps` are there so that the creator of the `Slot` could pass some props by default to the Fill, not in order to pass the `state`, `actions` or `libraries` down.

I guess we can "filter out" the props like: 

```
const Slot: SlotType = ({ name, state, children, libraries, actions, ...slotProps }) => {
	// We destructure the `libraries` and `actions` but not use them
	...
}
```

Reply:
I guess so. Yeah, think it's a good idea to filter out the props like that. :+1:

Reply:
Hey guys, just curious: why don't you use `useConnect` with `injectProps: false`? It was created specifically to  avoid this problem:

```ts
const Slot: SlotType = ({ name, children, ...slotProps }) => {
  const { state } = useConnect();

  // Get the data, either from props or the current link.
  const data = slotProps.data || state.source.get(state.router.link);

  // Get the fills for this name.
  const fills = useFills(name);

  return (
    <>
      {fills.length > 0
        ? fills.map(({ Fill, props, key }) => (
            <Fill key={key} name={key} data={data} {...slotProps} {...props} />
          ))
        : children}
    </>
  );
};

export default connect(Slot, { injectProps: false });
```

Reply:
You're right, this is the cleanest way :)